src/core/Akka/Actor/Props.cs

```diff
@@ -22,8 +22,76 @@ namespace Akka.Actor
     ///   private Props otherProps = props.WithDeploy(deployment info);
     ///  </code>
     /// </summary>
-    public class Props
+    public class Props : IEquatable<Props>
     {
+        public bool Equals(Props other)
+        {
+            if (ReferenceEquals(null, other)) return false;
+            if (ReferenceEquals(this, other)) return true;
+            return CompareDeploy(other) && CompareSupervisorStrategy(other) && CompareArguments(other) && CompareInputType(other) && CompareProducer(other);
+        }
+
+        private bool CompareInputType(Props other)
+        {
+            return true;
+            return inputType == other.inputType;
+        }
+
+        private bool CompareDeploy(Props other)
+        {
+            return true;
+            return Equals(Deploy, other.Deploy);
+        }
+
+        private bool CompareProducer(Props other)
+        {
+            return true;
+            return Equals(producer, other.producer);
+        }
+
+        private bool CompareSupervisorStrategy(Props other)
+        {
+            return true;
+            return Equals(SupervisorStrategy, other.SupervisorStrategy);
+        }
+
+        private bool CompareArguments(Props other)
+        {
+            return true;
+            if (other == null)
+                return false;
+
+            if (Arguments == null && other.Arguments == null)
+                return true;
+
+            if (Arguments == null)
+                return false;
+
+            return Arguments.SequenceEqual(other.Arguments);
```

Comment:
Definitely not the most performant way of doing it, but you could compare the serialized output of both objects?